File: Decorator/DecoratorCoffee/DecoratorCoffee/MilkDecorator.cs
namespace DecoratorCoffee
{
    public class MilkDecorator : BaseDecorator
    {
        public MilkDecorator(ICoffee coffee) : base(coffee)
        {
        }

        public override double GetCost() => base.GetCost() + 0.50;
        public override string GetDescription() => base.GetDescription() + ", milk";
    }
}

File: Decorator/Java/DecoratorCoffee/src/MilkDecorator.java
public class MilkDecorator extends BaseDecorator{
    public MilkDecorator(ICoffee coffee) {
        super(coffee);
    }

    @Override
    public double getCost() {
        return super.getCost() + 0.50;
    }

    @Override
    public String getDescription() {
        return super.getDescription() + ", milk";
    }
}
